doc/Dhcp_Mitigation/DHCP Mitigation.md

```diff
@@ -0,0 +1,252 @@
+
+
+### Requirements
+
+- Support for Linux traffic control (tc) for implementing the rate-limiting mechanism 
+- Support for CLI commands for configuring DHCP rate-limiting
+
+### Architecture Design
+The overall SONiC architecture will remain the same. Changes are made only in the SONiC Utilities container, SwSS container, and Config_DB.
+
+<figure style="text-align:center;">
+    <img src="../../images/dhcp_mitigation_hld/design_image.png" width="80%"/>
+    <figcaption>Figure 4: Feature Design</figcaption>
+</figure>
+
+### Sequence Diagram to add rate-limit
+
+<figure style="text-align:center;">
+    <img src="../../images/dhcp_mitigation_hld/sequence_diagram_add5.drawio (1).png" width="80%"/>
+    <figcaption>Figure 5: Sequence diagram illustrating the process of deleting rate-limit for DHCP DoS mitigation</figcaption>
+</figure>
+
+
+### Sequence Diagram to Del rate-limit 
+
+<figure style="text-align:center;">
+    <img src="../../images/dhcp_mitigation_hld/deleting_sequence.drawio.png" width="80%"/>
+    <figcaption>Figure 6: Sequence diagram illustrating the process of deleting rate-limit for DHCP DoS mitigation</figcaption>
+</figure>
+
+### Configuration &  Management
+#### CONFIG_DB 
+
+New attribute “dhcp_rate_limit” will be added to “PORT_TABLE” to support DHCP mitigation rate.
+
+ 
+      { "PORT":{
+       "Ethernet0": 
+       {
+      "admin_status": "up",
+    	"alias": "Ethernet1/1",
+    	"asic_port_name": "Eth0-ASIC0",
+    	"description": "ARISTA01T2:Ethernet3/1/1",
+    	"lanes": "33,34,35,36",
+    	"mtu": "9100",
+    	"pfc_asym": "off",
+    	"role": "Ext",
+    	"speed": "40000",
+      "dhcp_rate_limit" : "300"    
+         }
+      }}
+
+### SAI API 
+
+No SAI API change or addition is needed for this HLD. 
+
+
+### CLI/YANG model Enhancements 
+
+##### CLI  Configuration Commands
+
+Proposed SONiC CLI commands (sonic-utilities)
+-   config interface dhcp-mitigation-rate add [port] [packet-rate]
+-   config interface dhcp-mitigation-rate delete [port] [packet-rate]
+-   show interface dhcp-mitigation-rate
+
+Background Linux TC commands
+-   sudo tc qdisc add dev [interface] handle ffff: ingress
+-   sudo tc filter add dev [interface] protocol ip parent ffff: prio 1 u32 match ip protocol 17 0xff match ip dport 67 0xffff police rate [byte-rate] burst [byte-rate] conform-exceed drop
+
+
+##### YANG Model
+
+New leaf “dhcp_rate_limit” will be added to support DHCP mitigation rate.
+
+     leaf dhcp_rate_limit {
+   	        description: "dhcp rate limit ;
+   				  type uint32 {
+   					   range 0..8000;
+   				 }
+   			 }
+
+
+### Testing Requirements/Design  
```

Comment:
Where will the test cases to check functionality be defined? Here there are only CLI test definitions

Reply:
Hi @Yarden-Z , yes the test cases mentioned here are for CLI which will be implemented in the sonic-utilities repo. Other than these, we plan on contributing test plans and test case code to the sonic-mgmt repo for functionality testing of this feature.